crates/ra_ide_api/src/inlay_hints.rs

```diff
@@ -104,7 +104,15 @@ fn get_pat_type_hints(
         .map(|(range, pat_type)| InlayHint {
             range,
             kind: InlayKind::TypeHint,
-            label: pat_type.display(db).to_string().into(),
+            label: pat_type
+                .fold(&mut |ty| match ty {
+                    Ty::Apply(ApplicationTy { ctor: TypeCtor::Closure { .. }, .. }) => Ty::Unknown,
+                    _ => ty,
+                })
+                .display(db)
+                .to_string()
+                .replace("{unknown}", "_")
```

Comment:
In addition to the commented above, it would be great to leave the `{unknown}` without any changes: currently it's a very easy to notice annotation that shows that RA type inference does not work for the case hence a good motivation to create an issue or fix the problem.

Reply:
Yeah, `?` would be better I think, it's shorter :)

Reply:
I could replace it with `?` here, but it's hard to leave existing `{unknown}`'s in place without using a more elaborate setup. So I'll leave this part for a future PR.